crates/ruma-client-api/src/room/get_summary.rs

```diff
@@ -0,0 +1,100 @@
+//! `GET /_matrix/client/v1/summary/{roomIdOrAlias}`
+//!
+//! Experimental API enabled with MSC3266.
+//!
+//! Returns a short description of the state of a room, with state events.
+
+pub mod msc3266 {
+    //! `MSC3266` ([MSC])
+    //!
+    //! [MSC]: https://github.com/deepbluev7/matrix-doc/blob/room-summaries/proposals/3266-room-summary.md
+
+    use ruma_common::{
+        api::{request, response, Metadata},
+        metadata,
+        room::RoomType,
+        space::SpaceRoomJoinRule,
+        EventEncryptionAlgorithm, OwnedMxcUri, OwnedRoomAliasId, OwnedRoomId, OwnedRoomOrAliasId,
+        OwnedServerName, RoomVersionId,
+    };
+    use ruma_events::room::member::MembershipState;
+
+    const METADATA: Metadata = metadata! {
+        method: GET,
+        rate_limited: false,
+        authentication: AccessTokenOptional,
+        history: {
+            unstable => "/_matrix/client/unstable/im.nheko.summary/rooms/:room_id_or_alias/summary",
+        }
+    };
+
+    /// Request type for the `summary/room_id_or_alias` endpoint.
+    #[request(error = crate::Error)]
+    pub struct Request {
+        /// Alias or ID of the room to be summarized.
+        #[ruma_api(path)]
+        pub room_id_or_alias: OwnedRoomOrAliasId,
+
+        /// An optional list of servers the invited homeserver should attempt to peek at the room.
```

Comment:
The type is not optional here so it's a bit weird that the docs say it. I'd change it to something like:

```suggestion
        /// A list of servers the homeserver should attempt to use to peek at the room.
        ///
        /// Defaults to an empty `Vec`.
```